src/db/commit.rs

```diff
@@ -277,10 +300,153 @@ pub enum FromHashError {
     LoadCommitFailed(LoadError),
 }
 
+#[cfg(test)]
```

Comment:
Why not put all this stuff in the `test` mod defined below?

Reply:
i broke it out into a separate test-only module instead